test/document.test.js

```diff
@@ -12538,6 +12538,38 @@ describe('document', function() {
     assert.equal(purchaseFromDbCart.products[0].product.name, 'Bug');
     assert.equal(purchaseFromDbCart.singleProduct.name, 'Bug');
   });
+
+  it('handles virtuals that are stored as objects but getter returns string with toJSON (gh-14446)', async function() {
+    const childSchema = new mongoose.Schema();
+
+    childSchema.virtual('name')
+      .set(function(values) {
+        for (const [lang, val] of Object.entries(values)) {
+          this.set(`name.${lang}`, val);
+        }
+      })
+      .get(function() {
+        return this.$__getValue(`name.${this.lang}`);
+      });
+
+    childSchema.add({ name: { en: { type: String }, de: { type: String } } });
+
+    const ChildModel = db.model('Child', childSchema);
+    const ParentModel = db.model('Parent', new mongoose.Schema({
+      children: [childSchema]
+    }));
+
+    const child = await ChildModel.create({ name: { en: 'Stephen', de: 'Stefan' } });
+    child.lang = 'en';
```

Comment:
from what i can tell `lang` is not actually part of the schema, is this required for this issue or should it be added?

Reply:
I think that's by design, it's meant to be a property that isn't saved to the db. Could make it into a virtual, but I don't think that's strictly necessary for this test.